staging/src/k8s.io/component-base/config/serializer/encode.go

```diff
@@ -0,0 +1,260 @@
+	if !ok {
+		return []byte{}, fmt.Errorf("encoding format not supported: %s", format)
+	}
+	// versionSpecificEncoder writes out YAML bytes for exactly this group version
+	versionSpecificEncoder := cs.codecs.EncoderForVersion(encoder, gv)
+	// Encode the object to YAML for the given version
+	return runtime.Encode(versionSpecificEncoder, obj)
+}
+
+// EncodeMany encodes multiple inputs in a gvk-object map into a byte slice where the objects are encoded as multiple YAML documents
+func (cs *configSerializer) EncodeMany(format EncodingFormat, objs DecodedConfigs) ([]byte, error) {
+	docs := EncodedConfigs{}
+	for gvk, obj := range objs {
+		b, err := cs.Encode(format, gvk.GroupVersion(), obj)
+		if err != nil {
+			return nil, err
+		}
+		if err := docs.Add(gvk, b); err != nil {
+			return nil, err
+		}
+	}
+	return docs.GetAllDocuments(), nil
+}
+
+// EncodingFormat is a high-level, typed constant configuring how an object should be encoded
+type EncodingFormat string
+
+const (
+	// YAML encodes a config file in the YAML format
+	YAML EncodingFormat = "YAML"
+	// JSON encodes a config file in compact JSON format
+	JSON EncodingFormat = "JSON"
+	// JSONPretty encodes a config file in pretty JSON format
+	JSONPretty EncodingFormat = "JSONPretty"
+)
+
+// DecodedConfigs represents a collection of decoded configuration files.
+// The objects are indexed by their GroupVersionKind.
+type DecodedConfigs map[schema.GroupVersionKind]runtime.Object
+
+// EncodedConfigs represents a collection of byte-encoded configuration files.
+// The different byte slices are indexed by their GroupVersionKind. The kind name
+// is unique per set.
+type EncodedConfigs map[schema.GroupVersionKind][]byte
+
+// Add adds a new file to the set, but enforces that kind name is unique across the set
+func (ec *EncodedConfigs) Add(gvk schema.GroupVersionKind, b []byte) error {
+	// Enforce that kind is unique across this
+	for existinggvk := range *ec {
+		if existinggvk.Kind == gvk.Kind {
+			return fmt.Errorf("kind %q is specified twice in YAML file", gvk.Kind)
+		}
+	}
+	(*ec)[gvk] = b
+	return nil
+}
+
+// GetAllDocuments returns all documents in the set, in alphabetical order based on their kind,
+// separated by the YAML document separator.
+func (ec *EncodedConfigs) GetAllDocuments() []byte {
+	kinds := []string{}
+	kindMap := map[string]schema.GroupVersionKind{}
+	sortedContent := [][]byte{}
+	for gvk := range *ec {
+		kinds = append(kinds, gvk.Kind)
+		kindMap[gvk.Kind] = gvk
+	}
+	sort.Strings(kinds)
+	for _, kind := range kinds {
+		sortedContent = append(sortedContent, (*ec)[kindMap[kind]])
+	}
+
+	return bytes.Join(sortedContent, []byte(YAMLDocumentSeparator))
+}
+
+// ReadConfigSource reads a source string, which is a comma-delimited string that can point to multiple
+// configuration files, or directories with JSON or YAML files. Every source file may contain multiple
+// YAML documents
+func ReadConfigSource(srcString string) (EncodedConfigs, error) {
```

Comment:
Why do we need this to parse a string?
Wouldn't this information typically either already be formatted as a list in structured data, or have been parsed into a list by the flag parser?